### PR TITLE
Allow job concurrency to be set in config

### LINKED ISSUE #1108: Add possibility to set default number of concurrent jobs in .cargo/config
I usually prefer to run `cargo build -j 1` because `cargo build` makes the rest of my system very slow during the process.

I think it would be nice if there was an option to set the default value of this option in the configuration file.

### PR DESCRIPTION
**Pull request by cmr**
Closes #1108

### CODE REVIEW

File: src/cargo/util/config.rs [hunk 1]
```diff
@@ -147,6 +159,7 @@ impl<E, S: Encoder<E>> Encodable<S, E> for ConfigValue {
 impl ConfigValue {
     fn from_toml(path: &Path, toml: toml::Value) -> CargoResult<ConfigValue> {
         match toml {
+            toml::Value::Integer(val) => Ok(CV::Integer(val as f64, path.clone())),
```
Comment (alexcrichton) [inline src/cargo/util/config.rs]: f64?

File: src/cargo/util/config.rs [hunk 2]
```diff
@@ -30,6 +30,13 @@ impl<'a> Config<'a> {
     pub fn new(shell: &'a mut MultiShell,
                jobs: Option<uint>,
                target: Option<string::String>) -> CargoResult<Config<'a>> {
+        let config_jobs = match get_config(try!(os::getcwd()), "jobs").integer() {
+            Ok(jobs) if jobs.to_uint().is_some() => Some(jobs as uint),
+            Ok(..) => return Err(human("jobs from config is negative or too big")),
+            Err(e) if e.is_human() => None,
```
Comment (alexcrichton) [inline src/cargo/util/config.rs]: What is this case for

**Comment by alexcrichton** [comment]
Thanks! Could you also add some words to `src/doc/config.md`?

(travis build is also failing)

**Comment by alexcrichton** [comment]
I ended up rolling this into #1172, thanks!

**Comment by cmr** [comment]
Whoops, slipped my mind. Thanks!

On Thu, Jan 15, 2015 at 3:21 AM, Alex Crichton <notifications@github.com>
wrote:

> I ended up rolling this into #1172
> <https://github.com/rust-lang/cargo/pull/1172>, thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/cargo/pull/1114#issuecomment-70053325>.
>



-- 
http://octayn.net/

**Pull request closed without merging**